packages/boot/src/booters/datasource.booter.ts

```diff
@@ -24,12 +24,13 @@ import {BootBindings} from '../keys';
  * @param projectRoot - Root of User Project relative to which all paths are resolved
  * @param bootConfig - DataSource Artifact Options Object
  */
+@bind({tags: {configPath: 'datasources'}})
```

Comment:
I find it rather unusual to use `tags` as a mean for providing `configPath` option. My understanding so far was that `tags` are used to group bindings into tagged groups.

Let's find a better solution that will allow `@bind()` users to specify extra metadata.

I'd prefer the following:

```suggestion
@bind({configPath: 'datasources'})
```

If it would be too difficult to allow arbitrary top-level properties in `@bind` object, then let's introduce a new property dedicated to custom options, e.g.

```suggestion
@bind({options: {configPath: 'datasources'}})
```

Reply:
Either way, if this requires changes in `@bind` decorator, then please make those changes in a new pull request, so that we can review them in isolation.

Reply:
> I find it rather unusual to use tags as a mean for providing configPath option. My understanding so far was that tags are used to group bindings into tagged groups.

Tags are not only used for grouping but also for metadata. We already use tags such as `name`, `key`, and `namespace` to customize how a binding is created.

Reply:
> > I find it rather unusual to use `tags` as a mean for providing `configPath` option. My understanding so far was that `tags` are used to group bindings into tagged groups.
> >
> > Let's find a better solution that will allow `@bind()` users to specify extra metadata.
>
> Tags are not only used for grouping but also for metadata. We already use tags such as `name`, `key`, and `namespace` to customize how a binding is created.

Well, `name`, `key` and `namespace` make sense as tags to me, similarly to how physical products can be tagged with an EAN barcode and a price tag.

However, `configPath` seems different to me. It's not a metadata used to customize how a binding is created, it's a metadata specifying configuration options for other parts of our IoC system.

I feel we should separate those two kinds of metadata.

@strongloop/loopback-maintainers @strongloop/loopback-next thoughts?

Reply:
The `configPath` is now changed to `artifactNamespace`. We use it to infer how the configuration for a given booter can be resolved against boot options.